【Heng Jee Kuan] iP

hengjeekuan opened the pull request:
### JeeniusPro

> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

JeeniusPro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ _SUPER FAST_ to use

All you need to do is.

1. download it from [here](https://github.com/hengjeekuan/ip/tree/A-Jar)
2. double click it
3. add your tasks 
4. let it manage for you 😉 

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

hengjeekuan committed:
Edited Duke.java

hengjeekuan committed:
Changed all occurrences of Duke to Jeenius

hengjeekuan committed:
Added Echo function

hengjeekuan committed:
Added ability to store text and list them

hengjeekuan committed:
Added ability to mark tasks as done

hengjeekuan committed:
Created methods instead to prevent repetitive code

hengjeekuan committed:
Added Deadline and Event class

hengjeekuan committed:
Added Todos, Events and Deadlines class

hengjeekuan committed:
Added TextUiTesting

hengjeekuan committed:
Added all error handling

hengjeekuan committed:
Added delete method

hengjeekuan committed:
Added ability to create data folder if doesn't exist

hengjeekuan committed:
Added method to save tasks

hengjeekuan committed:
Add save tasks method into all methods that edit list

hengjeekuan committed:
Add save function

hengjeekuan committed:
Merge branch 'branch-Level-7'

hengjeekuan committed:
Fix bugs

hengjeekuan committed:
Update event and deadline task

hengjeekuan committed:
Add dates and time

hengjeekuan committed:
Add Jeenius.txt

hengjeekuan committed:
Merge branch 'branch-Level-8'

hengjeekuan committed:
Extract out code into more classes

hengjeekuan committed:
Move java files into respective packages

hengjeekuan committed:
Merge remote-tracking branch 'origin/add-gradle-support'

nictjh commented on src/main/java/jeenius/command/Parser.java:
```diff
@@ -0,0 +1,92 @@
+            ui.printTaskList(tasks.getTasks());
+        } else if (input.startsWith("todo")) {
+            String[] parts = input.split(" ", 2);
+            if (parts.length < 2) throw new JeeniusException("bro how do you todo nothing??? ADD A DESCRIPTION FOR YOUR TODO");
```
The conditional should be put on a separate line, could help in future debugging if necessary :>!
Other than that LGTM!!

pastchum replied:
Line is too long, can be difficult to read. consider splitting into 2 lines.

nictjh commented on src/main/java/jeenius/storage/Storage.java:
```diff
@@ -0,0 +1,66 @@
+                boolean isDone = parts[1].trim().equals("1");
+                String description = parts[2].trim();
+
+                switch (taskType) {
```
The case clauses shouldn't have any indentation as per coding standards. Can try to configure IDE to standardise the style

nictjh commented on src/main/java/jeenius/task/Task.java:
```diff
@@ -0,0 +1,37 @@
+        return this.description;
+    }
+
+    public Boolean getDone() {
```
This getter function can follow the "use a prefix such as is, has, was, etc. for boolean variables and methods". (Small things man give chance need fill quota)

nictjh commented on src/main/java/jeenius/Jeenius.java:
```diff
@@ -0,0 +1,56 @@
+            loadedTasks = new TaskList(storage.load());
+        } catch (JeeniusException e) {
+            ui.printError(e.getMessage());
+            loadedTasks = new TaskList(new ArrayList<>());
```
The Generic here could be filled with the super class (Task) for clarity, readability, and type safety. Prevent any inference made by Java.

nictjh commented on src/main/java/jeenius/list/TaskList.java:
```diff
@@ -0,0 +1,36 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class TaskList {
```
LGTM man

nictjh commented on src/main/java/jeenius/task/Deadline.java:
```diff
@@ -0,0 +1,28 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
```
LGTM!

nictjh commented on src/main/java/jeenius/task/Event.java:
```diff
@@ -0,0 +1,35 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Event extends Task {
```
LGTM bro

nictjh commented on src/main/java/jeenius/ui/Ui.java:
```diff
@@ -0,0 +1,40 @@
+
+import java.util.List;
+
+public class Ui {
```
Good job man, there's really nothing to nitpick at all

nictjh reviewed (commented):
Good work man, learnt alot from ya

pastchum commented on src/main/java/jeenius/Jeenius.java:
```diff
@@ -0,0 +1,56 @@
+                ui.printError(e.getMessage());
+            }
+        }
+    }
```
Good naming conventions, good code

pastchum commented on src/main/java/jeenius/command/Parser.java:
```diff
@@ -0,0 +1,92 @@
+            } catch (Exception e) {
+                throw new JeeniusException("Failed to mark/unmark. Use: mark/unmark [task number]");
+            }
+        } else {
```
I like the way you format the code in each "if" case. Very readable and neat, except for the few lines that seem to be very long and can be split into 2.

pastchum commented on src/main/java/jeenius/list/TaskList.java:
```diff
@@ -0,0 +1,36 @@
+    public Task getTask(int index) {
+        return tasks.get(index);
+    }
+
```
Should name method getSize to follow naming standard for methods. Apart from that LGTM

pastchum commented on src/main/java/jeenius/ui/Ui.java:
```diff
@@ -0,0 +1,40 @@
+
+public class Ui {
+
+    public void printWelcomeMessage() {
```
Good use of already defined functions to neaten and reduce repeated code. LGTM

pastchum reviewed (commented):
Very good code, apart from few minor nits this code was very well done!

hengjeekuan committed:
Add JUnit

hengjeekuan committed:
Merge tag 'A-JUnit'

hengjeekuan committed:
Push branches

hengjeekuan committed:
Add A-Jar

hengjeekuan committed:
Merge branch 'branch-A-Jar'

hengjeekuan committed:
Add JavaDoc comments to Task.java

hengjeekuan committed:
Change method name to follow naming standards for methods

hengjeekuan committed:
Revert "Change method name to follow naming standards for methods"

This reverts commit 2ca36b77a874162180e9ddda00d6f1ac5eee6876.

hengjeekuan committed:
Change method name to follow naming standard for methods

hengjeekuan committed:
Add Find feature

hengjeekuan committed:
Add JavaDoc to Todo.java, Deadline.java and Event.java

hengjeekuan committed:
Add JavaDoc for Parser.java

hengjeekuan committed:
Add Javadoc for TaskList.java

hengjeekuan committed:
Add JavaDoc to Storage.java

hengjeekuan committed:
Add JavaDoc comments to Jeenius.java, JeeniusException.java and Ui.java

hengjeekuan committed:
Add A-CodingStandard

hengjeekuan committed:
Merge branch 'branch-A-JavaDoc'

hengjeekuan committed:
Merge branch 'branch-A-CodingStandard'

hengjeekuan committed:
Merge branch 'branch-Level-9'

hengjeekuan committed:
Fix checkstyles error

hengjeekuan committed:
Add dependencies in build.gradle

hengjeekuan committed:
Add up to Part 2 for JavaFX

hengjeekuan committed:
Add Iteration 1 of Part 3

hengjeekuan committed:
Add ScrollPane

hengjeekuan committed:
Add Iteration 2 of Part 3

hengjeekuan committed:
Add Iteration 3 of Part 3

hengjeekuan committed:
Add FXML

hengjeekuan committed:
Add GUI

hengjeekuan committed:
Merge branch 'branch-Level-10'

hengjeekuan committed:
Add assertions into Tasklist.java

Assertions have been added to Tasklist.java to document important assumptions and prevent potential runtime errors. These assertions serve to verify that critical conditions hold throughout the program execution reducing chances of unexpected behaviour.

Let's:
- Add an assertion in the constructor to ensure the task list is never null after initialization
- Add an assertion in 'findTasks()' to prevent a 'NullPointerException' when processing a search keyword
- Add an assertion in 'deleteTask()' and 'getSize()' to validate that the provided index is within bounds before performing operations
- Add an assertion in 'addTask()' to prevent null tasks from being added

hengjeekuan committed:
Add assertions to Parser.java to validate preconditions and input constraints

Assertions have been added to Parser.java to ensure that critical conditions
are met before executing commands. These assertions help catch logical errors
early during development and debugging, improving the robustness of the parser.

Let's:
- Add assertions to validate that `input`, `tasks`, `ui`, and `storage`
  are not null before processing commands.
- Ensure that `deleteTask()`, `mark/unmark`, and `getSize()` only operate on
  valid task indices, preventing out-of-bounds errors.
- Add assertions in `findTasks()` and task creation commands (`todo`,
  `deadline`, `event`) to verify that input strings are not null or empty.
- Enforce correct splitting of parameters in `deadline` and `event` tasks
  to prevent index errors.

hengjeekuan committed:
Add assertions to Jeenius.java to enforce valid initialization and input handling

Assertions have been added to Jeenius.java to ensure that essential components
are properly initialized and critical input validations are enforced. These
assertions help catch logical errors early, preventing unintended behavior.

Let's
- Add assertions to validate that `filePath` is not null or empty before initializing storage.
- Ensure that `ui`, `storage`, `parser`, and `tasks` are not null after initialization.
- Add assertions in `run()` to verify that the `ui` and `parser` are ready before execution.
- Includ an assertion in `getResponse()` to ensure the input string is never null.
- Verifiy that user input in `run()` is not null before processing.

hengjeekuan committed:
Merge pull request #2 from hengjeekuan/branch-A-Assertions

Branch-A-Assertions

hengjeekuan committed:
Improve code quality

hengjeekuan committed:
Merge pull request #3 from hengjeekuan/branch-A-CodeQuality

Improve code quality

hengjeekuan committed:
Edit code so that GUI that opens up is running Jeenius.java

hengjeekuan committed:
Merge branch 'GUI-fix'

hengjeekuan committed:
Add C-Sort

hengjeekuan committed:
Merge branch 'branch-C-Sort'

hengjeekuan committed:
Improve GUI

hengjeekuan committed:
Set up product website

hengjeekuan committed:
Change README
